addon/services/viewport.js

```diff
@@ -66,5 +70,29 @@ export default Service.extend({
     if (this._globalWatcher) {
       this._globalWatcher.destroy();
     }
+  },
+
+  invalidate() {
+    spaniel.invalidate();
+  },
+
+  on(eventName, callback, root = window) {
```

Comment:
what does this API have to do with this service? it doesn't look like it belongs here at all.

Reply:
The intention was solely succinct abstract event binding methods to be called on the viewport. The use-case is to call `spaniel.invalidate()`. Im totally cool with removing `on/off/onDebouce` from the API (this PR) and rather just documenting the implementing leveraging existing API's.

Same answer for:
> there is already a run.debounce util in ember itself. also, this doesn't not use anything in the service and does not seem like it has to do with spaniel.

Reply:
lets pull on/off/onDebounce outside of this, just document suggested use of other apis. otherwise looks good to me